lib/chess/moves/generator.ex

```diff
@@ -1,35 +1,51 @@
 defmodule Chess.Moves.Generator do
-  @moduledoc false
+  @moduledoc """
+  Generates moves for a piece. We can generate moves in a straight line,
+  and these can be combined to make an entire pieces allowed moves. Or we
+  can generate moves based on a pattern of vectors. The knight works
+  like this.
+  """
+
+  alias Chess.Board
+
+  # `movement` is either a vector (bishop, rook, queen)
+  # or a pattern (king, knight)
+  def moves(board, {file, rank}, movement) do
+    board
+    |> Board.piece({file, rank})
+    |> Map.get("colour")
+    |> _moves(board, {file, rank}, movement)
+  end
 
   # Move generation for pieces that move in straight lines
-  def moves(_colour, _board, {0, _rank}, {-1, _}), do: []
-  def moves(_colour, _board, {_file, 0}, {_, -1}), do: []
-  def moves(_colour, _board, {7, _rank}, {1, _}), do: []
-  def moves(_colour, _board, {_file, 7}, {_, 1}), do: []
-  def moves(colour, board, {file, rank}, {fv, rv}) do
+  def _moves(_colour, _board, {0, _rank}, {-1, _}), do: []
+  def _moves(_colour, _board, {_file, 0}, {_, -1}), do: []
+  def _moves(_colour, _board, {7, _rank}, {1, _}), do: []
+  def _moves(_colour, _board, {_file, 7}, {_, 1}), do: []
+  def _moves(colour, board, {file, rank}, {fv, rv}) do
```

Comment:
What's the underscore for? Is it not used? Do you mean for it to be private?

Reply:
I might not have a popular opinion, but I'm not much of a fan of the underscore for Elixir to hide the recursive part of the function. There's likely a better name for the function, like in the fizzbuzz example: `defp generate_word` instead of `defp _fizzword`.

For this, what do you think of `defp moves_for_player` instead of `defp _moves`? I'm all for following convention though, so it's up to you.